examples/file_dialog.py

```diff
@@ -0,0 +1,17 @@
+"""FileDialog with magicgui."""
+from pathlib import Path
+from magicgui import event_loop, magicgui
+
+
+# may also add Qt-style filter to filename options:
+# e.g. {"filter": "Images (*.tif *.tiff)"}
+@magicgui(filename={"mode": "existing_file"})
```

Comment:
Didn't we agree to use 'mode': 'r' for this?

Reply:
Other than that the code looks really good! Like that's it, that's my full review! :joy:

Reply:
yeah I liked that, @GenevieveBuckley, can you make that change?  and instead of "files", make it a `List[Path]` type hint?  might run into some issues there, but let me know

Reply:
I'm sorry @tlambert03, what do you mean by "instead of "files", make it a List[Path] type hint?". Where in the code should I be looking?

Reply:
@jni I'd prefer not to replace Talley's names, but instead add `'r'` and '`w'` as aliases (because my brain finds it confusing to remember one mixed set of naming conventions but is a-ok with having an optional alternative way to specify the same thing)

```python
class FileDialogMode(Enum):
    """FileDialog mode options.

    EXISTING_FILE - returns one existing file.
    EXISTING_FILES - return one or more existing files.
    OPTIONAL_FILE - return one file name that does not have to exist.
    EXISTING_DIRECTORY - returns one existing directory.

    r - read mode, returns one existing file. Alias of EXISTING_FILE.
    w - write mode, returns one file name that does not have to exist.
        Alias of OPTIONAL_FILE.
    """

    EXISTING_FILE = "getOpenFileName"
    EXISTING_FILES = "getOpenFileNames"
    OPTIONAL_FILE = "getSaveFileName"
    EXISTING_DIRECTORY = "getExistingDirectory"
    # Aliases
    r = "getOpenFileName"
    w = "getSaveFileName"
```

Reply:
I'm ok with aliases, but (a) I would definitely prefer for 'r' to be the "promoted" use, and (b) I'd go for `r='getOpenFileNames'`. btw, does that allow selecting of directories? I think it does iirc.

Reply:
I'm happy for 'r'/'w' to be the promoted use too @jni

Sadly no, `'getOpenFileNames'` does not allow selecting directories. There isn't an option available that will do that.